### DISCUSSION TITLE
Re-think UI, navigation & layout

### DISCUSSION THREAD

**Discussion by martpie** [martpie/museeks#640] [Ideas]
Playing with #639, I'm thinking of going to a more traditional navigation, with header at the top, navigation on the left.

Pros:

- Jump to any playlist in one click from the main library view
- Labels will make navigation clearer for newcomers
- We fix some space issues we have on the header to fix #351

Cons: 

- The main view taking 100% of the width is very neat, and avoid distractions
- Less space available to show songs information

Example of the new layout:

<img width="1012" alt="Screenshot 2022-05-18 at 15 19 06" src="https://user-images.githubusercontent.com/1311607/169048143-93c0954e-9938-4ea8-9601-d67a31451660.png">

Update 28-05-2022:

<img width="1012" alt="Screenshot 2022-05-28 at 07 36 37" src="https://user-images.githubusercontent.com/1311607/170811743-a54a58a1-4a23-4d2d-80ff-2ac1ec4cedf4.png">

Update 31-05-2022 (contrast is too low, but the main idea is here):

<img width="1012" alt="Screenshot 2022-05-31 at 15 53 27" src="https://user-images.githubusercontent.com/1311607/171190342-438a1e68-76ef-468d-9663-2f994260eea6.png">

I am looking for feedback here: should it be solved in another way? Do you prefer it as it or do you like this new layout? Any feedback is welcome!

**Comment by martpie** [comment]
On a sidenote, I am also thinking on compacting the tracks list more. If this is controversial, I may provide an option on the interface to switch between "regular"/"compact", again, feedback welcome!

### Regular

<img width="1012" alt="Screenshot 2022-05-28 at 07 42 27" src="https://user-images.githubusercontent.com/1311607/170811901-c569ea82-9d96-4092-aba3-07330081a7a1.png">

### Compact

<img width="1012" alt="Screenshot 2022-05-28 at 07 42 20" src="https://user-images.githubusercontent.com/1311607/170811905-bb550f0e-8f7b-4f93-b1a8-e80c74635d04.png">

**Reply by jvegaf** [reply]
what do you think about this 

![image](https://user-images.githubusercontent.com/1075386/171053562-d607aca8-8616-4730-a404-d7ac3e73ecaa.png)

**Reply by martpie** [reply]
That's the idea indeed, but I'm not working on any UI for now, just UX/layout. I'm re-writing large parts of the app right now.

**Reply by jvegaf** [reply]
Ok, tell me how can I help you coding any part

**Reply by blackshibe** [reply]
Could "regular" show the album artwork in a bigger size than the current Museeks window? 
Alternatively, show the album thumbnail in the list itself. It's a rare feature to see but I miss it from Spotify.

**Comment by logchan** [comment]
Some thoughts...

- Vertical list of playlists is pretty nice
- Reduce the horizontal margins of the sidebar in compact mode
- Keep the search bar in header for potentially more space, also makes the following possible:
- Toggle the sidebar to show icons (or initials of playlists) instead, for larger horizontal space

Instead of having one compact mode toggle, I'd make it "style presets" of different configurations (of margins) that users can choose from.

One could go further to allow adding custom configurations in an advanced tab.

**Comment by jangorecki** [comment]
Do you know GPUI?
Did you evaluate its feasibility to handle UI in museeks?

https://www.gpui.rs/

_*I am not UI dev_

**Reply by martpie** [reply]
Rust UI libraries are for now out of the equation because of their lack of support for accessibility.

Also, GPUI is too unstable and Zed's folks recommend to not use it except for tinkering as it may evolve due to Zed's needs, and not other users needs.

**Comment by PeacheyMcKeitch** [comment]
Just installed Museeks in my never ending quest to find a mac/linux music player that isnt an over complicated mess like most of them these days!

Glad to say Museeks fits the bill pretty well. Personally my two cents is I would love to see an artists colum as well as the sidebar with library/playlists ect. A bit like the modern versions of apple music has. 

Nice way to quickly get to the arist without searching. An album colum would be overkill though imo.

**Reply by martpie** [reply]
Coming soon :) this week probably.
